kie-dmn-feel/src/main/java/org/kie/dmn/feel/util/EvalHelper.java

```diff
@@ -285,4 +291,106 @@ public static String lcFirst(final String name) {
         return name.toLowerCase().charAt( 0 ) + name.substring( 1 );
     }
 
+    /**
+     * Compares left and right operands using the given predicate and returns TRUE/FALSE accordingly
+     *
+     * @param left
+     * @param right
+     * @param ctx
+     * @param op
+     * @return
+     */
+    public static Boolean compare(Object left, Object right, EvaluationContext ctx, BiPredicate<Comparable, Comparable> op) {
+        if ( left == null || right == null ) {
+            return null;
+        } else if ( (left instanceof Period && right instanceof Period ) ) {
+            // periods have special compare semantics in FEEL as it ignores "days". Only months and years are compared
+            Period lp = (Period) left;
+            Period rp = (Period) right;
+            Integer l = lp.getYears() * 12 + lp.getMonths();
+            Integer r = rp.getYears() * 12 + rp.getMonths();
```

Comment:
Just by the "static" look of it, it seems the Integer stands for the month-based-equivalent of the period.
Therefore if I do if I do as per the current existing commit,
```
Say you have a periodA as left of 1year and 2 month = 1y*12+2m = 14 month-based-equivalent period
Say you have a periodB as right of 2year and 1 month = 2y*12+1m = 25 month-based-equivalent period.
```
This implying, periodA is smaller than periodB.
So if I pass "greater then" as a BiPredicate-of-Comparable as function, l=14, r=25, and would return NO, which is the expected result indeed.

If I do as you suggest "Isn't month * 12 +year correct?" would imply.
```
periodA = 2m*12+1y = 25
periodB = 1m*12+2y = 14
```
and I pass "greater then" as a BiPredicate-of-Comparable as function, l=25, r=14, and would return yes, but that is not correct as I think we can agree with the above premise.

Kindly let me know, please?